File: DSA/Ayush_Mishra_590014844/Day_19_Question_3.c
#include <stdio.h>
#include <stdbool.h>

bool search_matrix(int matrix[][100], int rows, int cols, int target) {
    int r = 0, c = cols - 1;
    while (r < rows && c >= 0) {
        if (matrix[r][c] == target)
            return true;
        else if (matrix[r][c] > target)
            c--;
        else
            r++;
    }
    return false;
}

int main() {
    int rows, cols;
    printf("Enter rows and columns: ");
    scanf("%d %d", &rows, &cols);
    int matrix[100][100];
    printf("Enter matrix:\n");
    for (int i = 0; i < rows; i++)
        for (int j = 0; j < cols; j++)
            scanf("%d", &matrix[i][j]);
    int target;
    printf("Enter target: ");
    scanf("%d", &target);
    printf("Found: %s\n", search_matrix(matrix, rows, cols, target) ? "true" : "false");
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_19_Question_3.py
def search_matrix(matrix, target):
    if not matrix: return False
    rows, cols = len(matrix), len(matrix[0])
    r, c = 0, cols - 1
    while r < rows and c >= 0:
        if matrix[r][c] == target:
            return True
        elif matrix[r][c] > target:
            c -= 1
        else:
            r += 1
    return False

rows = int(input("Enter number of rows: "))
matrix = []
for i in range(rows):
    matrix.append(list(map(int, input(f"Enter row {i+1}: ").split())))
target = int(input("Enter target: "))
print("Found:", search_matrix(matrix, target))
